components/camel-hdfs/src/main/java/org/apache/camel/component/hdfs/HdfsConfiguration.java

```diff
@@ -171,33 +186,35 @@ private String getString(Map<String, Object> hdfsSettings, String param, String
 
     private List<HdfsProducer.SplitStrategy> getSplitStrategies(Map<String, Object> hdfsSettings) {
         List<HdfsProducer.SplitStrategy> strategies = new ArrayList<>();
-        for (Object obj : hdfsSettings.keySet()) {
-            String key = (String) obj;
-            if ("splitStrategy".equals(key)) {
-                String eit = (String) hdfsSettings.get(key);
-                if (eit != null) {
-                    String[] strstrategies = eit.split(",");
-                    for (String strstrategy : strstrategies) {
-                        String tokens[] = strstrategy.split(":");
-                        if (tokens.length != 2) {
-                            throw new IllegalArgumentException("Wrong Split Strategy " + key + "=" + eit);
-                        }
-                        HdfsProducer.SplitStrategyType sst = HdfsProducer.SplitStrategyType.valueOf(tokens[0]);
-                        long ssv = Long.valueOf(tokens[1]);
-                        strategies.add(new HdfsProducer.SplitStrategy(sst, ssv));
-                    }
+
+        splitStrategy = getString(hdfsSettings, "splitStrategy", kerberosNamedNodes);
+
+        if (isNotEmpty(splitStrategy)) {
+            String[] strstrategies = splitStrategy.split(",");
```

Comment:
nit: do you think this makes sense?
```suggestion
            String[] strStrategies = splitStrategy.split(",");
```

Reply:
Makes sense.
I'll do another pull request for refactoring and other issues found by sonar.